src/base_directories.rs

```diff
@@ -300,7 +336,11 @@ impl BaseDirectories {
         // This crate only supports Unix, and the behavior of `std::env::home_dir()` is only
         // problematic on Windows.
         #[allow(deprecated)]
-        let home = std::env::home_dir();
+        let home = if home.as_os_str().is_empty() {
+            std::env::home_dir()
+        } else {
+            Some(PathBuf::from(home))
+        };
```

Comment:
This was the bulk of the change, which was pretty straightforward.

Reply:
Important to note that `std::env::home_dir()` doesn't use the `env_var()` argument that's passed in. I mention this in the tests down below.